src/org/zaproxy/zap/extension/pscanrules/TestInfoSessionIdURL.java

```diff
@@ -191,21 +193,49 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
     	                        msg);
     	
     	                parent.raiseAlert(id, alert);
-    	                
-    	        	    // Now try to check if there exists a referer inside the content
-    	                // i.e.: There is an external link for which 
-    	                // a referer header would be passed including this session token
-    	                try {
-    	                    checkSessionIDExposure(msg, id);
-    	                } catch (URIException e) {
-    	                }
     	                // We don't break on this one.
     	                // There shouldn't be more than one per URL but bizarre things do happen.
     	                // Improbable doesn't mean impossible.
+    	                found = true;
             		}
             	}
             }
-        }     
+        }
+        if (!found) {

```

Comment:
`&& msg.getRequestHeader().getURI().getEscapedPath() != null` to avoid a potential NPE.

Reply:
Oops sorry you did mentioned that earlier, fixed.